posts.showと_post_commnets.html.erbの修正

msgyu opened the pull request:
# 何か？
- posts.showのメソッドが重複しており、エラーが発生。一方を削除して改善
```
  def show
    @post = Post.find(params[:id])
    @comments = @post.comments.includes(:user).order("created_at DESC")
    @comment = @post.comments.build(user_id: current_user.id, post_id: @post.id) if current_user
  end

  def show
  end
```
- show.html.erbの記述がすべて_post_comments.html.erbに移行されたが、部分テンプレートの意味がないため、コメントに関するhtmlだけ残し、他はshow.html.erbに移行させた。

変更前はこの1行のみ

```show.html.erb
<%= render 'shared/post_comments' %> 
```

変更後はcommentの箇所にrenderを配置

```show.html.erb
          </ul>
          <% if user_signed_in? && @post.user_id == current_user.id %>
            <div class="post_detail__bottom_box__admin_actions">
              <%= link_to "編集", edit_post_path(@post.id) %>
              <%= link_to "削除", @post, method: :delete, data: { confirm: '本当に削除して良いですか?', cancel: 'やめる', commit: '削除する'}, title: '削除確認' %>
            </div>
          <% end %>
        </div>
      </div>
    </div>
    <%= render 'shared/post_comments' %>
  </div>
</div> 
```
## 変更されたファイル
- posts_controller.rb
- show.html.erb
- _post_comments.html.erb

msgyu committed:
重複したshowメソッドを削除

def show
end
が二つ存在し、記載が空が下にあるため、記述が打ち消されてたため、これを削除

msgyu committed:
set_postをbefore_actionに追加

@postは他のメソッドでも利用する共通部分なため、before_actionで定義。重複した記述をなくした

msgyu committed:
.comment_areaの記述以外は、show.html.erbに移行。

show.html.erbの記述すべてを_post_comments.html.erbに移行していたため、部分テンプレートの強みを活かせていなかった。そのため、.comment_area配下のhtmlを残し、それ以外はshow.html.erbに移行した。

tattu501 commented:
LGTM

masato-sa93 commented on app/controllers/posts_controller.rb:
```diff
@@ -1,21 +1,17 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, except: [:index, :show]
   before_action :authenticate_admin!, only: [:edit, :update, :destroy]
+  before_action :set_post, except: [:index, :new]
```
`except: [:index, :new]`だと、create時にもset_postされて重複になってしまうので、expectにcreateを追加するかonlyに変えましょうー！

msgyu replied:
exceptにcreateを追加しました

masato-sa93 commented on app/controllers/posts_controller.rb:
```diff
@@ -1,21 +1,17 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, except: [:index, :show]
   before_action :authenticate_admin!, only: [:edit, :update, :destroy]
+  before_action :set_post, except: [:index, :new]
```
また、`authenticate_admin!`時にも@postのインスタンスを定義しているので、edit, update, destroy時にも重複になっているので、こちらも修正お願いしますー！

msgyu replied:
重複箇所を削除

masato-sa93 reviewed (changes requested):
レビュー完了しましたー！
postコントローラーが複雑になりそうなので、ご確認と修正をお願いします！

msgyu committed:
set_post, except: [:index, :new, :create]に更新

優先度が高いため、before_action :set_postを一番はじめに配置。
createメソッドも不要なため、追加
また、authenticate_admin!の重複箇所を削除

masato-sa93 reviewed (approved):
LGTMですー！
